### ISSUE TITLE
Create a browser bookmarklet for quick-adding recipes

### ISSUE THREAD

**Issue by paulburton0** [nextcloud/cookbook#431]
A bookmarklet would make saving recipes to Cookbook easy without the "heavyweight" development overhead of creating a full-blown API or a browser extension.

**Comment by christianlupus** [comment]
You mean as an alternative to #288?
The overhaul is necessary for quick development anyway. Also, there is the issue that the bookmarklet runs in the context of the *owning* site so has no login credentials to NC as far as I understand your suggestion.

So please be a bit more specific, what this issue is about. Is it a suggestion? Is it a question if you can open a PR :smile:?

**Comment by paulburton0** [comment]
I'm not a coder per se, so I guess this is a suggestion/feature request.

I don't mean that a browser bookmarklet would replace the suggested "batch add" functionality in. Rather, it would streamline the saving of a recipe. 

Perhaps a use-case scenario would help: If I'm browsing, say NY Times recipes and I see one I want to add to my NC Cookbook, while I'm on that recipe page, I click the bookmarklet (saved conveniently to my browser's bookmarks toolbar) and it's added without having to copy the URL, open a new tab, go to Cookbook, paste the URL, and click the button to save it to Cookbook.

Again, I'm not up on all of the security/architectural implications of this, but I do know that the Nextcloud Bookmarks app has a bookmarklet that works similar to my scenario above (i.e. when you're on a page that you want to save to your Nextcloud bookmarks app, you execute the bookmarklet while on that page and it "just does it").

**Comment by geeseven** [comment]
It has been many years since I have used bookmarklets.  At the time, I found them super useful.  

As @paulburton0 mentioned, the Bookmark app has a [bookmarklet].  For some reason, I can't seem to add it to any browser.

[bookmarklets]: https://github.com/nextcloud/bookmarks/blob/master/src/components/ViewBookmarklet.vue

**Comment by Bugsbane** [comment]
> the bookmarklet runs in the context of the owning site so has no login credentials to NC

The News app used to have a bookmarklet which was generated in its settings area. The user dragged it to their bookmarks toolbar, and then clicking on it would add the URL to the News app itself. The exact same thing would be amazing (you could even almost straight steal the code!) just putting the URL into the Cookbook app instead.

From memory, the way it dealt with the security side was that you had to currently be logged in to Nextcloud using that browser for it to work.

**Comment by christianlupus** [comment]
You are right and I see another benefit for this: Pages loading parts of their content in a lazy fashion could be parsed better. We would let the client browser do the loading and heavy lifting and just parse the DOM and forward the result to the cookbook app. So, this is for sure on the task list.
